plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergRegisterTableProcedure.java

```diff
@@ -0,0 +1,420 @@
+    {
+        PARQUET("parquet"),
+        ORC("orc"),
+        AVRO("avro"),
+        /**/;
+
+        private final String name;
+
+        IcebergFileFormat(String format)
+        {
+            this.name = format;
+        }
+
+        @Override
+        public String toString()
+        {
+            return this.name;
+        }
+    }
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        metastoreDir = Files.createTempDirectory("test_iceberg_register_table").toFile();
+        metastore = createTestingFileHiveMetastore(metastoreDir);
+        fileSystem = new HdfsFileSystemFactory(HDFS_ENVIRONMENT).create(TestingConnectorSession.SESSION);
+        return IcebergQueryRunner.builder()
+                .setMetastoreDirectory(metastoreDir)
+                .setIcebergProperties(ImmutableMap.of(REGISTER_TABLE_PROCEDURE_ENABLED, "true"))
+                .build();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+            throws IOException
+    {
+        deleteRecursively(metastoreDir.toPath(), ALLOW_INSECURE);
+    }
+
+    @DataProvider
+    public static Object[][] fileFormats()
+    {
+        return Stream.of(IcebergFileFormat.values())
+                .map(icebergFileFormat -> new Object[] {icebergFileFormat})
+                .toArray(Object[][]::new);
+    }
+
+    @Test(dataProvider = "fileFormats")
+    public void testRegisterTableWithTableLocation(IcebergFileFormat icebergFileFormat)
+    {
+        String tableName = "test_register_table_with_table_location_" + icebergFileFormat + "_" + randomTableSuffix();
+
+        assertUpdate(format("CREATE TABLE %s (a int, b varchar, c boolean) with (format = '%s')", tableName, icebergFileFormat));
+        assertUpdate(format("INSERT INTO %s values(1, 'INDIA', true)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s values(2, 'USA', false)", tableName), 1);
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')");
+
+        String tableLocation = getTableLocation(tableName);
+        String showCreateTableOld = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+
+        // Drop table from hive metastore and use the same table name to register again with the metadata
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL iceberg.system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        String showCreateTableNew = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+
+        assertUpdate(format("INSERT INTO %s values(3, 'POLAND', true)", tableName), 1);
+        assertThat(query(format("SELECT * FROM %s", tableName)))
+                .matches("VALUES ROW(INT '1', VARCHAR 'INDIA', BOOLEAN 'true'), ROW(INT '2', VARCHAR 'USA', BOOLEAN 'false')," +
```

Comment:
nit: You could split the string to multiple lines for readability sake

Sample code:

https://github.com/trinodb/trino/blob/6e29d60fd99370e197739bf74e00cb20725bb4d4/plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java#L336-L338